File: app/src/main/java/sample/jbanse/demosql/data/dao/News.java
package sample.jbanse.demosql.data.dao;

import android.support.annotation.Nullable;

import com.google.auto.value.AutoValue;

import java.util.Date;

import sample.jbanse.demosql.data.db.DateAdapter;

/**
 * Created by julien on 20/09/2017.
 */
@AutoValue
public abstract class News implements NewsModel {

    public static final NewsModel.Factory<News> FACTORY = new NewsModel.Factory<News>(new NewsModel.Creator<News>() {

        @Override
        public News create(long id, @Nullable String title, Date publicationDate, int topRatedPosition, @Nullable Long authorId) {
            return new AutoValue_News(id, title, publicationDate, topRatedPosition, authorId);
        }
    }, new DateAdapter());
}

File: app/src/main/java/sample/jbanse/demosql/data/dao/News.kt
package sample.jbanse.demosql.data.dao

import com.google.auto.value.AutoValue

/**
 * Created by julien on 12/09/2017.
 */
@AutoValue
abstract class News : NewsModel {

    companion object {

        val FACTORY: NewsModel.Factory<News> = NewsModel.Factory<News>(NewsModel.Creator<News>(::AutoValue_News))

        //val NEWS_DETAIL_MAPPER :RowMapper<NewsDetail> = FACTORY.selectNewsDetailMapper(NewsModel.SelectNewsDetailCreator<News,Author, NewsDetail>(::AutoValue_NewsDetail), Author.FACTORY)
    }
}